src/gba/audio.c

```diff
@@ -161,6 +161,10 @@ void GBAAudioWriteSOUND2CNT_HI(struct GBAAudio* audio, uint16_t value) {
 }
 
 void GBAAudioWriteSOUND3CNT_LO(struct GBAAudio* audio, uint16_t value) {
+	if (!audio->enable) {
+		return;
+	}
```

Comment:
Actually, I realize that this check needs to be done _outside_ of the function since stuff gets written into the I/O memory buffer outside of this function. That probably is similar for all of the other audio writes...there's actually a TODO about this in GBIORead.

Reply:
Well, according to GBATEK:

> While Bit 7 is cleared, both PSG and FIFO sounds are disabled, and all PSG registers at 4000060h..4000081h are reset to zero (and must be re-initialized after re-enabling sound). However, registers 4000082h and 4000088h are kept read/write-able (of which, 4000082h has no function when sound is off, whilst 4000088h does work even when sound is off).

Reply:
It should be done as a separate commit at least, but same PR might be fine.